src/main/java/org/folio/rest/impl/TenantPermsAPI.java

```diff
@@ -392,8 +398,15 @@ public void postTenantpermissions(OkapiPermissionSet entity, Map<String, String>
     return savePermList(moduleId, new ArrayList<>(permList.keySet()), vertxContext, tenantId)
         .compose(v -> {
           List<Future> futures = new ArrayList<>(permList.size());
+          PostgresClient pgClient = PostgresClient.getInstance(vertxContext.owner(), tenantId);
           permList.keySet().forEach(okapiPerm -> {
+            String newPermName = okapiPerm.getPermissionName();
             permList.get(okapiPerm).forEach(replaced -> {
+              // add new permission name to all relevant sub permissions
+              String oldPermName = replaced.getPermissionName();
+              futures.add(Future.<RowSet<Row>>future(p -> pgClient.execute(connection,
+                  String.format(ADD_PERM_TO_SUB_PERMS, tenantId, newPermName, oldPermName, newPermName), p)));
```

Comment:
String.format should be replaced by the PostgresClient.execute method that takes `Tuple params`.